vertica_python/vertica/connection.py

```diff
@@ -541,6 +541,7 @@ def enable_ssl(self, raw_socket, ssl_options):
         else:
             err_msg = "SSL requested but not supported by server"
             self._logger.error(err_msg)
+            self.close_socket()
```

Comment:
This `close_socket()` function only works when the connection is established, enabling TLS is just before that step. So this line actually doesn't take effect.

Reply:
Ah okay. I originally had raw_socket.close() before seeing the close_socket function, but that makes sense. I'll close this PR